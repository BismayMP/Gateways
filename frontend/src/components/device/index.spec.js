import { cleanup, render, screen } from '@testing-library/react'
import Device from './index'
import renderer from 'react-test-renderer'

afterEach(cleanup)

const formatDate = (str) => {
  const date = new Date(str)
  return date.toDateString()
}

describe('Details component', () => {
  const device = {
    _id: '610878604df35a739fc63b0c',
    vendor: '1',
    status: 'online',
    date: '2021-08-02T22:57:36.105Z',
  }
  const onDelete = jest.fn((item) => null)
  const statusChange = jest.fn((item) => null)
  test('should match snapshot', () => {
    const component = renderer.create(
      <Device
        device={device}
        onStatusChange={statusChange}
        onDelete={onDelete}
      />,
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  test('All elements are in the dom', () => {
    render(
      <Device
        device={device}
        onStatusChange={statusChange}
        onDelete={onDelete}
      />,
    )
    const buttons = screen.getAllByRole('button')
    buttons.forEach((button) => {
      expect(button).toBeInTheDocument()
      button.click()
    })
    expect(onDelete).toHaveBeenCalledTimes(1)
    expect(statusChange).toHaveBeenCalledTimes(1)
    const deviceStatus = screen.getByText('online')
    expect(deviceStatus).toBeInTheDocument()
    const deviceVendor = screen.getByText('name: 1')
    expect(deviceVendor).toBeInTheDocument()
    const deviceDate = screen.getByText(
      `Created at: ${formatDate('2021-08-02T22:57:36.105Z')}`,
    )
    expect(deviceDate).toBeInTheDocument()
  })
})
