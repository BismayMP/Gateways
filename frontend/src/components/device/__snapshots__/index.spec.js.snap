// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details component should match snapshot 1`] = `
<div
  className="device"
>
  <p
    className="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
  >
    name: 
    1
  </p>
  <p
    className="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
  >
    Created at: 
    Mon Aug 02 2021
  </p>
  <div>
    <div
      className="MuiChip-root MuiChip-colorPrimary"
      onKeyDown={[Function]}
      onKeyUp={[Function]}
    >
      <span
        className="MuiChip-label"
      >
        online
      </span>
    </div>
  </div>
  <div
    className="device-btns"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-outlinedSizeSmall MuiButton-sizeSmall MuiButton-disableElevation"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        Change status
      </span>
    </button>
    <button
      aria-label="delete"
      className="MuiButtonBase-root MuiIconButton-root remove-device-btn MuiIconButton-sizeSmall"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiIconButton-label"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-colorSecondary MuiSvgIcon-fontSizeSmall"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
          />
        </svg>
      </span>
    </button>
  </div>
</div>
`;
